packages/components/src/components/badge/badge.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import classNames from 'classnames';
+
+type TBadgeSize = 'medium' | 'large';
+type TBadgeSpacing = 'tight' | 'loose';
+type TWeight = 'bold' | 'regular';
+type TBackgroundColor = 'blue' | 'orange' | 'red' | 'gray';
+type TRoundedCorners = 4 | 2;
+
+interface Badge extends React.HTMLAttributes<HTMLSpanElement> {
+    className?: string;
+    color?: string;
+    label: string;
+    rounded_corners?: TRoundedCorners;
+    size?: TBadgeSize;
+    spacing?: TBadgeSpacing;
+    weight?: TWeight;
+}
+
+interface BadgeContained extends Badge {
+    type: 'contained';
+    background_color: TBackgroundColor;
+}
+
+interface BadgeBordered extends Badge {
+    type: 'bordered';
+}
+
+type BadgeProps = BadgeContained | BadgeBordered;
+
+const Badge = (props: BadgeProps) => {
+    const {
+        className,
+        color,
```

Comment:
I would suggest to name it differently
```suggestion
        custom_color,
```